src/server.ts

```diff
@@ -57,6 +60,15 @@ export class RecordReplayServer {
         return;
       }
 
+      if (
+        this.preventConditionalRequests &&
+        (req.method === "GET" || req.method === "HEAD")
+      ) {
+        // Headers are always coming in as lowercase.
+        delete req.headers["if-modified-since"];
+        delete req.headers["if-none-match"];
```

Comment:
Question: do we need to remove the headers from https://nodejs.org/api/http.html#messagerawheaders as well? I assume not if we're only forwarding `req.headers` elsewhere.

Reply:
Good question. I don't believe so as we're copying the headers out of `req` to pass upstream later on.